File: windows/Scanner/Rules/AESMCP009.cs
using AIExposureScanner.Scanner.Models;

namespace AIExposureScanner.Scanner.Rules;

public sealed class AESMCP009 : IRule
{
    public string Id => "AES-MCP-009";
    public Severity Severity => Severity.Critical;

    public IReadOnlyList<Finding> Evaluate(ScanFacts facts) =>
        facts.McpServers
            .Where(server => !server.Disabled)
            .Where(server => AiThreatPatterns.IsRemoteInstallScript(server.Command, server.Args))
            .Select(server => this.Finding(server.AppId, server.Name, affectedPath: server.ConfigPath))
            .ToArray();
}

File: macos/Scanner/Sources/Scanner/Rules/AESMCP009.swift
public struct AESMCP009: Rule {
    public let id = "AES-MCP-009"
    public let severity: Severity = .critical

    public init() {}

    public func evaluate(_ facts: ScanFacts) -> [Finding] {
        facts.mcpServers.compactMap { server in
            guard !server.disabled else {
                return nil
            }
            guard AiThreatPatterns.isRemoteInstallScript(command: server.command, args: server.args) else {
                return nil
            }
            return finding(app: server.appId, serverName: server.name, affectedPath: server.configPath)
        }
    }
}
